File: app/src/main/java/com/example/myappcontacts/business/contactslist/ContactsListInteractor.java
package com.example.myappcontacts.business.contactslist;

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel;
import com.example.myappcontacts.data.repositories.contactslist.IContactsListRepository;

import java.util.List;
import java.util.UUID;

import io.reactivex.Completable;
import io.reactivex.Single;

public class ContactsListInteractor implements IContactsListInteractor {

    private IContactsListRepository mContactsListRepository;

    public ContactsListInteractor(IContactsListRepository contactsListRepository) {
        mContactsListRepository = contactsListRepository;
    }

    @Override
    public Single<UUID> addContact() {
        return mContactsListRepository.addContact();
    }

    @Override
    public Completable deleteContact(UUID contactId) {
        return mContactsListRepository.deleteContact(contactId);
    }

    @Override
    public Single<List<ContactsModel>> loadContactsList() {
        return mContactsListRepository.loadContactsList();
    }
}

File: app/src/main/java/com/example/myappcontacts/business/contactslist/ContactsListInteractor.kt
package com.example.myappcontacts.business.contactslist

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel
import com.example.myappcontacts.data.repositories.contactslist.IContactsListRepository
import java.util.UUID

import io.reactivex.Completable
import io.reactivex.Single
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

class ContactsListInteractor(private val contactsListRepository: IContactsListRepository) : IContactsListInteractor {

    override fun addContact(): Single<UUID> {
        return contactsListRepository.addContact()
    }

    override fun deleteContact(contactId: UUID): Completable {
        return contactsListRepository.deleteContact(contactId)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
    }

    override fun loadContactsList(): Single<List<ContactsModel>> {
        return contactsListRepository.loadContactsList()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
    }
}
